dev/benchmarks/platform_channels_benchmarks/lib/main.dart

```diff
@@ -0,0 +1,174 @@
+// Copyright 2014 The Flutter Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+import 'dart:math' as math;
+import 'dart:typed_data';
+
+import 'package:flutter/material.dart';
+import 'package:flutter/services.dart';
+import 'package:flutter_test/flutter_test.dart';
+
+import 'package:microbenchmarks/common.dart';
+
+List<Object> _makeTestBuffer(int size) {
+  final List<Object> answer = <Object>[];
+  for (int i = 0; i < size; ++i) {
+    switch (i % 9) {
+      case 0:
+        answer.add(1);
+        break;
+      case 1:
+        answer.add(math.pow(2, 65));
+        break;
+      case 2:
+        answer.add(1234.0);
+        break;
+      case 3:
+        answer.add(null);
+        break;
+      case 4:
+        answer.add(<int>[1234]);
+        break;
+      case 5:
+        answer.add(<String, int>{'hello': 1234});
+        break;
+      case 6:
+        answer.add('this is a test');
+        break;
+      case 7:
+        answer.add(true);
+        break;
+      case 8:
+        answer.add(Uint8List(64));
+        break;
+    }
+  }
+  return answer;
+}
+
+Future<void> _runTests() async {
+  assert(false,
+      "Don't run benchmarks in checked mode! Use 'flutter run --release'.");
+  const int numMessages = 10000;
+
+  const BasicMessageChannel<Object> channel = BasicMessageChannel<Object>(
+      'dev.flutter.echo.basic.standard', StandardMessageCodec());
+  final Stopwatch watch = Stopwatch();
+
+  watch.start();
+  for (int i = 0; i < numMessages; ++i) {
+    await channel.send(1234);
+  }
+  watch.stop();
+  final double smallPayloadTime = watch.elapsedMicroseconds / numMessages;
+
+  watch.reset();
+  final List<Object> largeBuffer = _makeTestBuffer(1000);
+  const StandardMessageCodec codec = StandardMessageCodec();
+  final ByteData data = codec.encodeMessage(largeBuffer);
+  print('Large buffer size: ${data.lengthInBytes}');
```

Comment:
remove the print?

Reply:
This was intentional to leave this in for the logs since it's difficult to make the datastructure of a specific size and it's important that its size is known in order to make valid comparisons.  I might actually tweak the size before I turn the tests on.  I gotta check to make sure this is what we want to test.

Reply:
Can we somehow capture this in the benchmark metric? Like, maybe make it `us/byte` or something? Or just include the number of bytes processed somewhere in the metrics?

Reply:
Even if we capture `us/byte` changing the size of the buffer in the test would create a change in `us/byte`, it won't scale linearly (byte bytes are copied 4 different times and serializing/deserializing isn't equal for all datatypes regardless of byte size).  We have to make sure when I turn this on, we never change the size or contents of the buffer.

Reply:
> I gotta check to make sure this is what we want to test.

The print alone doesn't tell me what the size should be, though, if I later come in and modify the benchmark. Ideally, this would have been expressed as an `assert(data.lengthInBytes == expectation)`. But asserts are disabled here. Maybe just throw if it doesn't have the expected size?

Reply:
I removed the print statement and instead added a comment that warns about ever editing that line of code.  I don't think there is anything else we can do or assert.